maps/randomvaults/objects.dm

```diff
@@ -649,3 +657,207 @@
 
 		CO.face_atom(AM)
 		CO.visible_message("<span class='notice'>\The [CO] looks at \the [AM] and hisses angrily!</span>")
+
+
+/* Spy Sat stuff */
+
+/obj/machinery/power/magtape_deck
+	name = "magnetic tape drive"
+	desc = "A primitive way of storing information. Used because of its longevity over most digital counterparts."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "blackbox"
+
+	active_power_usage = 500
+	density = 1
+	var/powered
+
+/obj/machinery/power/magtape_deck/New()
+	..()
+	connect_to_network()
+
+/obj/machinery/power/magtape_deck/process()
+	if(stat & BROKEN)
+		return
+	if(avail(active_power_usage))
```

Comment:
This probably doesn't do what you think it does, `avail` takes no parameters. What's this supposed to do? Looks like you could do without the `powered` var entirely as well